Resources/Catalog/RucioFileCatalogClient.py

```diff
@@ -0,0 +1,590 @@
+
+
+  @checkCatalogArguments
+  def listDirectory(self, lfns, verbose = False):
+    """ Returns the result of __getDirectoryContents for multiple supplied paths
+    """
+    result = {'OK': True, 'Value': {'Successful': {}, 'Failed': {}}}
+
+    for lfn in lfns:
+      try:
+        did = self.__getDidsFromLfn(lfn)
+        # First need to check if it's a dataset or container
+        meta = self.client.get_metadata(did['scope'], did['name'])
+        if meta['did_type'] == 'CONTAINER':
+          if lfn not in result['Value']['Successful']:
+            result['Value']['Successful'][lfn] = {'Files': {}, 'Links': {}, 'SubDirs': {}}
+          for child in self.client.list_content(scope=did['scope'], name=did['name']):
+            childName = child['name']
+            if self.convertUnicode:
+              childName = str(childName)
+            if child['type'] in ['DATASET', 'CONTAINER']:
+              result['Value']['Successful'][lfn]['SubDirs'][childName] = {'Mode': 509}
+              if verbose:
+                result['Value']['Successful'][lfn]['SubDirs'][childName] = {'Status': '-',
+                                                                            'CreationDate': datetime.datetime(1970, 1, 1),
+                                                                            'ChecksumType': '',
+                                                                            'OwnerRole': 'None',
+                                                                            'Checksum': '',
+                                                                            'NumberOfLinks': -999,
+                                                                            'OwnerDN': 'None',
+                                                                            'Mode': 509,
+                                                                            'ModificationDate': datetime.datetime(1970, 1, 1),
+                                                                            'GUID': '',
+                                                                            'LastAccess': 0,
+                                                                            'Size': 0}
+            else:
+              result['Value']['Successful'][lfn]['Files'][childName] = {'Mode': 509}
+              if verbose:
+                pass
+        elif meta['did_type'] == 'DATASET':
+          file_dict = {}
+          for file_did in self.client.list_files(scope=did['scope'], name=did['name']):
+            guid = file_did['guid']
+            file_dict[file_did['name']] = guid
+            if guid:
+              file_dict[file_did['name']] = str(uuid.UUID(guid))
+          if lfn not in result['Value']['Successful']:
+            result['Value']['Successful'][lfn] = {'Files': {}, 'Links': {}, 'SubDirs': {}}
+          for rep in self.client.list_replicas([did, ]):
+            if rep:
+              name = rep['name']
+              if self.convertUnicode:
```

Comment:
Subject to the outcome of my other Unicode comment, this feels like it should be a function so you don't have to full the code with `if`. Or better yet always use `str` as I think `convertUnicode` is always True?